### DISCUSSION TITLE
Flowdeck no stable X and Y state estimate

### DISCUSSION THREAD

**Discussion by HeleMartin** [bitcraze/discussions#316] [Q&A]
Hello,

We have a flowdeck which gives no stable position X and Y. For example, if the drone is on a plain surface and doesn't moove, the X and Y vary a lot from 0 to 100 in some seconds. Consequently, the drone crashes at each take off.
Here, the corresponding console prints : 
```
SYS: ----------------------------
SYS: Crazyflie 2.1 is up and running!
SYS: Build 5:826a44b95f0e (2021.06 +5) MODIFIED
SYS: I am 0x383535303539470900330035 and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
DECK_CORE: 1 deck(s) found
DECK_CORE: Calling INIT on driver bcFlow2 for deck 0
ZR2: Z-down sensor [OK]
PMW: Motion chip id: 0x49:0xB6
IMU: BMI088: Using I2C interface.
IMU: BMI088 Gyro connection [OK].
IMU: BMI088 Accel connection [OK]
IMU: BMP388 I2C connection [OK]
ESTIMATOR: Using Kalman (2) estimator
CONTROLLER: Using PID (1) controller
MTR-DRV: Using brushed motor driver
SYS: About to run tests in system.c.
EEPROM: I2C connection [OK].
STORAGE: Storage check [OK].
IMU: BMI088 gyro self-test [OK]
DECK_CORE: Deck 0 test [OK].
SYS: Self test passed!
STAB: Wait for sensor calibration...
SYS: Free heap: 17664 bytes
STAB: Ready to fly.
Waiting for activation ...
ESTKALMAN: State out of bounds, resetting
ESTKALMAN: State out of bounds, resetting
ESTKALMAN: State out of bounds, resetting
ESTKALMAN: State out of bounds, resetting
ESTKALMAN: State out of bounds, resetting
ESTKALMAN: State out of bounds, resetting
ESTKALMAN: State out of bounds, resetting
ESTKALMAN: State out of bounds, resetting
ESTKALMAN: State out of bounds, resetting
ESTKALMAN: State out of bounds, resetting
```
Do you have an explanation ?

Thanks,

**Comment by krichardsson** [comment]
How does it behave if you lift it up a bit?
Sometimes it is hard for the flow sensor to get good information as it is so close to the ground when the Crazyflie is not flying. 

I get the impression that you have more Flow decks? If this is correct, do they behave differently?

**Comment by HeleMartin** [comment]
It has the same behaviour. 
For the other flowdecks, it is ok even so close to the ground.

**Reply by tobbeanton** [reply]
Could there be e.g. dirt on the lens? Has it always been this way? If so send us an email to support (at) bitcraze.io and we will help you out with a replacement.
